src/main/java/net/minestom/arena/game/mob/SpiderMob.java

```diff
@@ -1,159 +1,113 @@
+                    instance.setBlock(spawnAt, Block.COBWEB);
+                    cobwebs.add(spawnAt);
+                }
             }
 
-            this.stop = target == null;
-
-            if (!stop) {
-
-                // Attack the target entity
-                if (entityCreature.getDistance(target) <= range) {
-                    entityCreature.lookAt(target);
-                    if (!Cooldown.hasCooldown(time, lastHit, delay)) {
-                        // TODO Add animation for throwing web
-
-                        Pos pos = target.getPosition();
-                        Random random = ThreadLocalRandom.current();
-                        Instance instance = target.getInstance();
-                        if (instance == null) return;
-
-                        List<Pos> cobwebs = new ArrayList<>();
-                        for (int i = 0; i < 8; i++) {
-                            Pos spawnAt = pos.add(
-                                random.nextInt(-1, 1),
-                                random.nextInt(0, 2),
-                                random.nextInt(-1, 1)
-                            );
-
-                            if (instance.getBlock(spawnAt).isAir()) {
-                                instance.setBlock(spawnAt, Block.COBWEB);
-                                cobwebs.add(spawnAt);
-                            }
-                        }
-
-                        target.scheduler().buildTask(() -> {
-                            Instance cobwebInstance = target.getInstance();
-                            if (cobwebInstance == null) return;
-
-                            for (Pos cobweb : cobwebs) {
-                                cobwebInstance.setBlock(cobweb, Block.AIR);
-                            }
-                        }).delay(5, TimeUnit.SECOND).executionType(ExecutionType.ASYNC).schedule();
-
-                        this.lastHit = time;
-                    }
-                    return;
-                }
+            projectile.scheduler().buildTask(() -> {
+                Instance cobwebInstance = projectile.getInstance();
+                if (cobwebInstance == null) return;
 
-                // Move toward the target entity
-                Navigator navigator = entityCreature.getNavigator();
-                final var pathPosition = navigator.getPathPosition();
-                final var targetPosition = target.getPosition();
-                if (pathPosition == null || pathPosition.distance(targetPosition) > range) {
-                    if (this.cooldown.isReady(time)) {
-                        this.cooldown.refreshLastUpdate(time);
-                        navigator.setPathTo(targetPosition);
-                    }
+                for (Pos cobweb : cobwebs) {
+                    cobwebInstance.setBlock(cobweb, Block.AIR);
                 }
-            }
-        }
+            }).delay(5, TimeUnit.SECOND).executionType(ExecutionType.ASYNC).schedule();
```

Comment:
Why is it async?

Reply:
Doesn't work if its sync

Reply:
I assume because you are removing the entity just after, stopping the scheduler with it.

But still, async tasks should not be used for such reason.

Reply:
Okay, now it's scheduled in the instance itself.